framework/aster-frame/src/task/processor.rs

```diff
@@ -141,19 +144,19 @@ impl PreemptInfo {
     }
 
     fn incease_num_locks(&self) {
-        self.num_locks.fetch_add(1, Relaxed);
+        self.num_locks.fetch_add(1, AcqRel);
     }
 
     fn decrease_num_locks(&self) {
-        self.num_locks.fetch_sub(1, Relaxed);
+        self.num_locks.fetch_sub(1, AcqRel);
     }
 
     fn is_preemptive(&self) -> bool {
-        self.num_locks.load(Relaxed) == 0
+        self.num_locks.load(Acquire) == 0
     }
 
     fn num_locks(&self) -> usize {
-        self.num_locks.load(Relaxed)
+        self.num_locks.load(Acquire)
```

Comment:
Isn't the preempt count a CPU-local variable where the concurrency access will never occur? I'm not sure why we can't get a mutable reference to CPU-local variables so we can exclusively modify them:
```rust
    fn decrease_num_locks(&mut self) {
        self.num_locks -= 1;
    }
```

Reply:
Thinking deeper. It seems that before having exclusive access to CPU-local variables, we should first disable preempt (otherwise the current task can be preempted and scheduled on another CPU by accident). So it looks like a chicken and egg situation, and solving it seems to be beyond the scope of this PR.

However, I still don't think the fixes in this PR can make a difference. The precise memory order used to preempt the count can be `Relaxed`, since the preempt count is always accessed by a single CPU, and never by other CPUs.

Reply:
To have an exclusive access to CPU-local variables, `CpuLocal::borrow_with` may help by the preempt&IRQ guard at the beginnig.

Not closely related to this PR, will there be risks to replace the `AtomicUsize` with `Cell<usize>` along with `borrow_with`?

Reply:
> To have an exclusive access to CPU-local variables, CpuLocal::borrow_with may help by the preempt&IRQ guard at the beginnig.

The implementation of `CpuLocal` looks like only a placeholder. It needs to be implemented when supporting multiple processors.

> Not closely related to this PR, as for num_locks, will there be risks to replace the AtomicUsize with Cell<usize> along with borrow_with?

I'm not sure about the motivation. Is there any benefit?

`Cell` is [!Sync](https://doc.rust-lang.org/std/cell/struct.Cell.html#impl-Sync-for-Cell%3CT%3E), which means references cannot be shared across threads. This limitation leads to more powerful APIs, so we can get a mutable reference to the inner data.

In contrast, `AtomicUsize`  is [Sync](https://doc.rust-lang.org/std/cell/struct.Cell.html#impl-Sync-for-Cell%3CT%3E). Its references can be shared across threads, resulting in limited APIs. We can never get a mutable reference to the inner data.

As for CPU-local variables, they should not be accessed across threads, so using `Cell` seems to be fine. But I still think both `AtomicUsize` and `Cell` are somehow unnecessary, instead, we should have exclusive access thus a mutable reference `&mut usize` fits better.

More unfortunately, since `borrow_with` disables preemption at the beginning, it causes an infinite recursion if `borrow_with` is used while trying to disable preemption. So it's a chicken and egg situation.